docs/development/testing_guide.rst

```diff
@@ -1,219 +1,263 @@
+names of the form :file:`test_*.py`. For example, tests for
+`plasmapy.formulary.parameters` are located at
+:file:`plasmapy/formulary/tests/test_parameters.py` relative to the top
+of the package. Example code contained within docstrings is tested to
+make sure that the actual printed output matches what is in the
+docstring.
+
+Running tests
+=============
 
-.. _testing-guidelines-motivation:
+PlasmaPy's tests can be run in the following ways:
+
+1. Creating and updating a pull request on GitHub_.
+2. Running pytest_ from the command line.
+3. Running tox_ from the command line.
+4. Running tests from an `integrated development environment` (IDE).
+
+We recommend that new contributors perform the tests via a pull request
+on GitHub_. Creating a draft pull request and keeping it updated will
+ensure that the necessary checks are run frequently. This approach is
+also appropriate for pull requests with a limited scope. This advantage
+of this approach is that the tests are run automatically and do not
+require any extra work. The disadvantages are that running the tests on
+GitHub_ is often slow and that navigating the test results is sometimes
+difficult.
 
-Motivation
-==========
+We recommend that experienced contributors run tests either by using
+pytest_ from the command line or by using your preferred IDE.
+Using tox_ is an alternative to pytest_, but running tests with tox_ is
+typically slower than running tests with pytest_.
 
-Tests are vital for software reliability and maintainability.  Writing
-tests requires additional effort now, but saves considerable time in the
-long run.  Tests enable us to modify code and quickly discover when we
-introduce errors [1]_.  Tests also provide future contributors with
-examples of how functions and classes were originally intended to be
-used.
+Using GitHub
+------------
 
-Tests should be readable and maintainable.  Well-written tests are
-easier to understand and modify when the behavior of a function or
-method is changed.  Consequently, tests should be held to the same
-code quality standards as the rest of the package.
+The recommended way for new contributors to run PlasmaPy's full test
+suite is to `create a pull request`_ from your development branch to
+`PlasmaPy's GitHub repository`_. The test suite will be run
+automatically when the pull request is created and every time changes
+are pushed to the development branch on GitHub_.
 
-When bugs are discovered, they should be turned into test cases to
-prevent the bug from emerging again in the future [2]_.
+The following checks are performed with each pull request. The results
+of the checks are found near the end of the *Conversation* tab in each
+pull request. Most of these checks have been automated using `GitHub
+Actions`_. Checks that pass are marked with ✔️, while tests that fail
+are marked with ❌. Click on *Details* for information about why a
+particular check failed.
```

Comment:
```suggestion
particular check failed.

.. note::

   For first-time contributors, existing maintainers may need to manually
   enable your GitHub Action test runs. This is, believe it or not,
   indirectly caused by the invention of cryptocurrencies. Read more about this at
   https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks
  
```

Reply:
Argh, I forgot about that, which is surprising given all the bitcoin mining I've been doing on GitHub pull requests to astropy_helpers.